lib/get-jira-issues.js

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env node
+
+const core = require('@actions/core');
+const github = require('@actions/github');
+
+async function run() {
+  // Init inputs and context
+  const owner = core.getInput('repository')?.split('/')[0] || github.context.repo.owner;
+  const repo = core.getInput('repository')?.split('/')[1] || github.context.repo.repo;
+  const baseBranch = core.getInput('base_branch') || github.context.payload.pull_request.base.ref;
+  const targetBranch = core.getInput('target_branch') || github.context.payload.pull_request.head.ref;
+  const jiraPrefixes = core.getInput('jira_prefixes');
+  const githubToken = core.getInput('github_token');
+
+  // Init GitHub client
+  const octokit = github.getOctokit(githubToken)
+
+  // Construct regex to match issue numbers
+  const projectPrefixes = jiraPrefixes.split(',').map(prefix => prefix.trim());
+  const prefixRegexInsert = projectPrefixes.map(prefix => `[${prefix}]{${prefix.length}}`).join('|');
+  const issueRegex = new RegExp(`((${prefixRegexInsert})[-\\s]*\\d+)`, 'i');
+
+  // Get all commit messages between the two branches
+  console.log(`Getting commits between ${baseBranch} and ${targetBranch}`);
+  const { data: commitData } = await octokit.rest.repos.compareCommits({
+    owner,
+    repo,
+    base: baseBranch,
+    head: targetBranch,
+  });
+  const commitMessages = commitData.commits.map(commit => commit.commit.message)
```

Comment:
(improvement): This can be extracted into a separated method to improve readability.

Reply:
Since I would need to pass all of these args and the Octokit client, it might make more sense to not extract this one.